controllers/storageclient/interface.go

```diff
@@ -1,10 +1,18 @@
 // Package storageclient provides the storage client for the controllers package
 package storageclient
 
+import "github.com/iter8-tools/iter8/base/summarymetrics"
+
+// versionMetricSummary is a map of metric names to summary metric values
+type versionMetricSummary map[string]summarymetrics.SummaryMetric
+
 // Interface enables interaction with a storage entity
 // Can be mocked in unit tests with fake implementation
 type Interface interface {
-	// CreateMetric(name string, value interface{}) error
-	// GetMetric(name string) (interface{}, error)
-	// DeleteMetric(name string) (interface{}, error)
+	// WriteMetric is called by the A/B/n SDK gRPC API implementation (SDK for application clients)
+	WriteMetric(applicationName string, version string, user string, metricName string, metricValue float64) error
```

Comment:
```suggestion
	WriteMetric(applicationName string, version int, user string, metricName string, metricValue float64) error
```

Reply:
version number and timestamp can be treated as the actual version

Reply:
Later refinement: instead of using the timestamp, hash the metadata of the Kubernetes object